### PR TITLE
Add realistic hlrc request serialization test base class and

### PR DESCRIPTION
**Pull request by martijnvg**
change hlrc ccr request tests to use AbstractRequestTestCase base class.

This way the request classes are tested in a more realistic setting.
Note this change also adds a test dependency on xpack core module.

Similar to #39844 but then for hlrc request serialization tests.
Relates to #39745

### CODE REVIEW

File: client/rest-high-level/build.gradle
Comment (atorok) [inline client/rest-high-level/build.gradle]: This is a test dependency but I think it's still one that we have no precedent for. 
Maybe it would be better to have these tests somewhere in `x-pack/qa` ?

Comment (martijnvg) [reply client/rest-high-level/build.gradle]: If these tests were integration tests then I would agree. However since these tests are unit tests, I think they should stay next to the code that these tests are testing.

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/2

File: client/rest-high-level/build.gradle
Comment (hub-cap) [reply client/rest-high-level/build.gradle]: I think a test dep on xpack is perfectly fine WRT unit tests here. I agree w/ @martijnvg but thank you for having the eagle eye @atorok and bringing this up, just because it is definitely a major dependency.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java
Comment (hub-cap) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java]: should this be overridable?

File: client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java
Comment (hub-cap) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java]: is this the only named writeables we will need for these tests?

**Review by hub-cap — COMMENTED** [review]
SO much cleaner. Ive left one Q about the named writables, and this is approve-worthy pending an answer there.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java
Comment (martijnvg) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java]: We can make it overridable. However all the other parsing tests don't do this either and have the number of tests runs also fixed to 20. So I think for now this is ok?

File: client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java
Comment (martijnvg) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java]: I turns out we need no named xcontents, at least for now. So I will change this to use an empty registry. I like to be careful requiring the usage of named xcontent on the hlrc side for now.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java
Comment (jasontedor) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java]: Do we need this? Our tests run a bazillion times a day in CI. Is 20x a bazillion buying us anything?

Comment (martijnvg) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java]: > Do we need this? Our tests run a bazillion times a day in CI. Is 20x a bazillion buying us anything?

Good question. We do this in the other xcontent parsing tests too. I personally like like this during development, so that silly mistakes are caught before I merge into master. On the other hand, I can also run the tests with `-Dtests.iters=N` flag. Like you say we keep paying the price until these tests are no longer run in CI. So from that perspective it makes sense to remove the loop. If we do this here then we should also do this in the other xcontent parsing tests. wdyt?

File: client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java
Comment (hub-cap) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java]: agree. good call.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java
Comment (hub-cap) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java]: ++ im fine w/ removing. I develop using tests.iters quite typically on the hlrc unit tests.

Comment (martijnvg) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java]: I've removed the iterations in this base class and `AbstractResponseTestCase`: e5b1ca778702f9f068dc2819d311412b6c4accf8

Comment (jasontedor) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java]: >If we do this here then we should also do this in the other xcontent parsing tests. wdyt?

+1

**Review by hub-cap — APPROVED** [review]
:shipit:

**Pull request merged**
